cpp/core/jni/JniWrapper.cc

```diff
@@ -544,7 +544,12 @@ JNIEXPORT jlong JNICALL Java_org_apache_gluten_vectorized_ColumnarBatchOutIterat
   auto ctx = gluten::getRuntime(env, wrapper);
 
   auto it = ctx->objectStore()->retrieve<ResultIterator>(iterHandle);
-  return it->spillFixedSize(size);
+  if (it == nullptr) {
```

Comment:
`it == nullptr` is not that clear to indicate `it` is closed. Could we leverage some kind of closing flags? Probably from Java / Scala side?

Reply:
ok